airflow/api_fastapi/core_api/routes/ui/dags.py

```diff
@@ -47,12 +47,20 @@
     DAGWithLatestDagRunsCollectionResponse,
     DAGWithLatestDagRunsResponse,
 )
+from airflow.api_fastapi.core_api.security import (
+    ReadableDagsFilterDep,
+    requires_access_dag,
+)
 from airflow.models import DagModel, DagRun
 
 dags_router = AirflowRouter(prefix="/dags", tags=["Dags"])
 
 
-@dags_router.get("/recent_dag_runs", response_model_exclude_none=True)
+@dags_router.get(
+    "/recent_dag_runs",
+    response_model_exclude_none=True,
+    dependencies=[Depends(requires_access_dag(method="GET"))],
+)
```

Comment:
Do we need both access to dags as well as `("GET", DagAccessEntity.RUN)` ?

Reply:
right, missed this one

Reply:
just added